### DISCUSSION TITLE
The Future of Egress Extensibility

### DISCUSSION THREAD

**Discussion by kkeirstead** [dotnet/dotnet-monitor#2500] [Announcements]
The dotnet-monitor team is currently working on an egress extensibility model that will allow for third-party egress extensions (initially detailed in #1635). Extension developers will be able to create and deploy their own egress providers alongside dotnet-monitor. In addition, we will be creating a dedicated GitHub repo that will be the home of our future built-in cloud providers. The existing Azure egress provider will be moved to this repo, and we will avoid breaking changes when this transition occurs.

As of today, we have not yet established a timeline for this transition; more details will be provided in the coming months regarding how users and contributors will be able to take advantage of this functionality. If you have any questions or suggestions, you can leave a comment on this discussion and we’ll keep you up-to-date with our progress.

Related:
- #4006

**Comment by birojnayak** [comment]
@kkeirstead do you guys have a spec or doc/ some blob (:)) which can be shared , would be happy to see...

**Reply by kkeirstead** [reply]
Hi Biroj, thanks for your interest! I'll plan on sharing a doc/more info in the next week or so - I'll post it here when it's ready.

**Reply by kkeirstead** [reply]
Here's a working doc of our plans for egress extensibility - note that this may change going forward. [Egress Extensibility Plans (External) - V1.docx](https://github.com/dotnet/dotnet-monitor/files/10167422/Egress.Extensibility.Plans.External.-.V1.docx)
